### DISCUSSION TITLE
keybinding "l" not work

### DISCUSSION THREAD

**Discussion by realyukii** [jarun/nnn#1789] [Q&A] [answered]
I tried to preview file by pressing l, which suggests fallback options such as vi, less, and sh as mentioned in the [wiki page](https://github.com/jarun/nnn/wiki/Usage). However, pressing 'l' doesn't seem to initiate any action. Is there something I might have overlooked?

**Comment by N-R-K** [comment, marked as answer]
Not enough information to do anything. If you build `nnn` with debug logging enabled (`make O_DEBUG=1`) then it should output a log file at `/tmp/nnndbg` which might reveal something.

**Reply by realyukii** [reply]
I tried rebuild package with this command:

```
MAKEFLAGS="O_DEBUG=1" makepkg -si 
```

but there's no such log file in tmp dir:

```
$ cat /tmp/nnndbg
cat: /tmp/nnndbg: No such file or directory
```

**Reply by jarun** [reply]
It's generated when you run `nnn`. If it's not, please try to debug why. The relevant code is in `dbg.h`.

**Reply by realyukii** [reply]
okay, compiling directly with make instead of makepkg works, here's the log when I press l on a file:

```
ln 7047: newpath=/home/reyuki/nnn/src/nnn-4.9/README.md
ln 7065: sb.st_mode=33188
ln 2471: pid=116042
ln 6575: __func__=redraw
ln 6590: path=/home/reyuki/nnn/src/nnn-4.9
```

now it works but the opener is a chromium browser, why the default opener is browser? I haven't configure it to be chromium
is there any place to lookup what the default app was setting for nnn?

**Reply by N-R-K** [reply]
> what the default app was setting for nnn?

`nnn` doesn't have any defaults. It simply uses `xdg-open`. You can either change your xdg defaults https://wiki.archlinux.org/title/Xdg-utils#xdg-mime. Or use a custom opener script via `$NNN_OPENER`.

**Reply by realyukii** [reply]
thanks for the information @N-R-K !
